packages/pigeon/lib/dart_generator.dart

```diff
@@ -332,16 +336,17 @@ $resultAt != null
                 'binaryMessenger: binaryMessenger);',
               );
             });
-            final String messageHandlerSetter =
-                isMockHandler ? 'setMockMessageHandler' : 'setMessageHandler';
+            final String messageHandlerSetter = isMockHandler
+                ? '_testBinaryMessengerBinding!.defaultBinaryMessenger.setMockDecodedMessageHandler<Object?>(channel, '
+                : 'channel.setMessageHandler(';
             indent.write('if (api == null) ');
             indent.addScoped('{', '}', () {
-              indent.writeln('channel.$messageHandlerSetter(null);');
+              indent.writeln('${messageHandlerSetter}null);');
             }, addTrailingNewline: false);
             indent.add(' else ');
             indent.addScoped('{', '}', () {
               indent.write(
-                'channel.$messageHandlerSetter((Object? message) async ',
+                '$messageHandlerSetter(Object? message) async ',
```

Comment:
might be worth a comment somewhere calling out that messageHandlerSetter contains the `(`

Reply:
Renamed the variable name to make that clear.